packages/repository/src/repositories/relation.repository.ts

```diff
@@ -55,41 +56,55 @@ export interface HasManyRepository<Target extends Entity> {
   ): Promise<number>;
 }
 
+export interface BelongsToRepository<Target extends Entity> {
+  find(options?: Options): Promise<Target>;
+}
+
 export class DefaultHasManyEntityCrudRepository<
   TargetEntity extends Entity,
   TargetID,
   TargetRepository extends EntityCrudRepository<TargetEntity, TargetID>
 > implements HasManyRepository<TargetEntity> {
   /**
    * Constructor of DefaultHasManyEntityCrudRepository
-   * @param targetRepository the related target model repository instance
+   * @param targetRepositoryGetter the related target model repository instance
    * @param constraint the key value pair representing foreign key name to constrain
    * the target repository instance
    */
+
+  public targetRepositoryGetter: Getter<TargetRepository>;
+
   constructor(
-    public targetRepository: TargetRepository,
+    targetRepository: TargetRepository | Getter<TargetRepository>,
     public constraint: DataObject<TargetEntity>,
-  ) {}
+  ) {
+    this.targetRepositoryGetter =
+      typeof targetRepository === 'object'
+        ? ((() => Promise.resolve(targetRepository)) as Getter<
+            TargetRepository
+          >)
+        : targetRepository;
+  }
 
   async create(
     targetModelData: DataObject<TargetEntity>,
     options?: Options,
   ): Promise<TargetEntity> {
-    return await this.targetRepository.create(
+    return (await this.targetRepositoryGetter()).create(
```

Comment:
this is from the effect of trying to solve the circular dependency issue. As mentioned above, we could have a separate decorator so that we don't end up using a promisified version of the repository

Reply:
Can we not inject the repository instead using `@inject.getter()`? or add support for this via `@repository`?